packages/gatsby/src/redux/run-sift.js

```diff
@@ -49,25 +49,29 @@ const createFilterCacheKey = (typeNames, filter) => {
   return typeNames.join(`,`) + `/` + paths.join(`,`) + `/` + comparator
 }
 
-const prepareQueryArgs = (filterFields = {}) =>
-  Object.keys(filterFields).reduce((acc, key) => {
+function prepareQueryArgs(filterFields = {}) {
+  const filters = {}
+  Object.keys(filterFields).forEach(key => {
     const value = filterFields[key]
     if (_.isPlainObject(value)) {
-      acc[key === `elemMatch` ? `$elemMatch` : key] = prepareQueryArgs(value)
+      filters[key === `elemMatch` ? `$elemMatch` : key] = prepareQueryArgs(
+        value
+      )
     } else {
       switch (key) {
         case `regex`:
-          acc[`$regex`] = prepareRegex(value)
+          filters[`$regex`] = prepareRegex(value)
           break
         case `glob`:
-          acc[`$regex`] = makeRe(value)
+          filters[`$regex`] = makeRe(value)
           break
         default:
-          acc[`$${key}`] = value
+          filters[`$${key}`] = value
       }
     }
-    return acc
-  }, {})
+  })
+  return filters
```

Comment:
don't think there is anything wrong with this. Just curious why you changed from reduce to loop?

Reply:
Shrug. I think it's easier to read this way and I was already changing something.